src/evernode.c

```diff
@@ -333,94 +342,94 @@ int64_t hook(int64_t reserved)
+                    // Take the max audit from the config.
+                    uint16_t conf_max_audit;
+                    GET_CONF_VALUE(conf_max_audit, DEF_MAX_AUDIT, CONF_MAX_AUDIT, "Evernode: Could not set default state for max audit.");
+                    TRACEVAR(conf_max_audit);
 
-                    // If auditor id is not within (HASH_SIZE - 3) we are taking sha512 hashes.
-                    uint32_t eligible_count = HASH_SIZE - 3;
-                    uint32_t iterations = (auditor_id - 1) / eligible_count;
-                    uint32_t host_id;
-                    if (iterations > 0)
-                    {
-                        uint8_t seed_hash[HASH_SIZE] = {0};
-                        for (int i = 0; GUARD(iterations), i < iterations; ++i)
-                        {
-                            if (util_sha512h(SBUF(seed_hash), audit_pick_ptr, HASH_SIZE) < 0)
-                                rollback(SBUF("Evernode: Could not generate the sha512h hash of the moment seed."), 1);
-                            audit_pick_ptr = &seed_hash;
-                        }
-                        trace(SBUF("Auditor host pick buffer: "), audit_pick_ptr, HASH_SIZE, 1);
-                        uint32_t pick_index = (auditor_id - 1) % eligible_count;
-                        host_id = (UINT32_FROM_BUF(audit_pick_ptr + pick_index) % host_count) + 1;
-                    }
-                    else
+                    // Take the minimum redeem amount from the config.
+                    uint16_t conf_min_redeem;
+                    GET_CONF_VALUE(conf_min_redeem, DEF_MIN_REDEEM, CONF_MIN_REDEEM, "Evernode: Could not set default state for min redeem.");
+                    TRACEVAR(conf_min_redeem);
+
+                    uint32_t active_host_count = host_count;
+
+                    uint8_t *moment_seed_ptr = &moment_seed_buf[8];
+                    trace(SBUF("Moment seed: "), moment_seed_ptr, HASH_SIZE, 1);
+
+                    uint32_t pick_start_host_id = (UINT32_FROM_BUF(moment_seed_ptr) % active_host_count) + 1;
+                    uint32_t pick_start_auditor_id = (UINT32_FROM_BUF(moment_seed_ptr + 1) % auditor_count) + 1;
+
+                    uint32_t pick_idx = ((auditor_count + auditor_id) - pick_start_auditor_id) % auditor_count;
+                    uint32_t pick_count = MIN(CEIL(active_host_count, auditor_count), conf_max_audit);
+                    uint32_t pick_host_from = pick_start_host_id + (pick_idx * pick_count);
+                    pick_host_from = (pick_host_from > active_host_count) ? (pick_host_from % active_host_count) : pick_host_from;
+                    // If this is the last index, it might not get assigned "pick_count" hosts.
+                    if (pick_idx == auditor_count - 1)
+                        pick_count = pick_count - (((pick_idx + 1) * pick_count) % active_host_count);
+
+                    // Setup the outgoing txns for all the hosts.
```

Comment:
// Setup the outgoing txns for all checkCreates representing hosts assigned for this auditor.